core/script.runtime/src/kotlin/script/dependencies/resolvers.kt

```diff
@@ -32,6 +50,7 @@ interface DependenciesResolver : @Suppress("DEPRECATION") ScriptDependenciesReso
 
     sealed class ResolveResult {
```

Comment:
Talked to @udalov in person. We seem to agree that: 
```
data class ResolutionResult(
    val dependencies: ScriptDependencies?,
    val diagnostics: List<ScriptDiagnostic>
)
```
is simpler and covers all the required cases. It needs some documentation around what `dependencies == null` means but so do all other considered variations.